shell/platform/android/io/flutter/plugin/text/ProcessTextPlugin.java

```diff
@@ -0,0 +1,175 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.text;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.os.Build;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.embedding.engine.systemchannels.ProcessTextChannel;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.PluginRegistry.ActivityResultListener;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ProcessTextPlugin implements FlutterPlugin, ActivityAware, ActivityResultListener {
+  private static final String TAG = "ProcessTextPlugin";
+
+  @NonNull private final ProcessTextChannel processTextChannel;
+  @NonNull private final PackageManager packageManager;
+  @Nullable private ActivityPluginBinding activityBinding;
+  @NonNull private Map<Integer, ResolveInfo> resolveInfosById = new HashMap<Integer, ResolveInfo>();
+
+  @NonNull
+  private Map<Integer, MethodChannel.Result> requestsByCode =
+      new HashMap<Integer, MethodChannel.Result>();
+
+  public ProcessTextPlugin(@NonNull ProcessTextChannel processTextChannel) {
+    this.processTextChannel = processTextChannel;
+    this.packageManager = processTextChannel.packageManager;
+
+    this.cacheResolveInfos();
```

Comment:
I'm wondering if we can do this elsewhere because this will happen immediately when a `ProcessTextPlugin` is created, and that will happen regardless of whether or not this functionality is used. Would it make sense to be part of `queryTextActions()` or `processTextAction(...)`?

Reply:
Done